sections/discussion/discussion-chapter.tex

```diff
@@ -1,9 +1,153 @@
+The issue is that it will probably not change the top-$N$ list of recommended items, so instead one should use metrics such as Precision@N, Recall@N or F1@N to determine the value of the context variables.
+The interesting context variables to examine would be the ones that affect individual items or categories of items differently.
+That would result in a change in the order of preference when doing recommendations.
+RMSE and MAE may not be able to express this but a better or worse Precision@N, Recall@N, F1@N would be able to capture this.
+\\\\
+However the RMSE and MAE metrics can still be used when optimizing an algorithm as they give an indication of how far the predicted ratings are from the actual ratings.
+In terms of usage when evaluating if a method is useful in real-world applications, other metrics that look at the quality of the top-$N$ predictions are more useful as those recommendations are what the user will be interacting with.
+
+\subsubsection{The trade-off between precision and recall and the F1-measure}\label{subsub:precrectradeoff}
+A trade-off between precision and recall is expected as mentioned in \autoref{sec:evaluationmetrics}.
+This is due to their definitions: precision uses the length of the proposed top-$N$ list as the denominator while recall uses the number of total relevant items.
+This means that if the amount of relevant items increases, the algorithm is more likely to hit a relevant item in the recommendation, thus improving overall precision.
+However, an increasing number of relevant items penalizes recall, thus there is a trade-off.
+As such, simply looking at one of these metrics can be misleading, as the results can be influenced in this manner.
+We thus use F1 as a metric to combine these metrics and generate more comparable results between the methods.
+\autoref{graph:MLPrecRecF1}, \autoref{graph:YahooPrecRecF1} and \autoref{graph:CoMoDaPrecRecF1} report the results for all of these methods.
+Because of the trade-off, F1 is the most interesting metric for comparison.
+
+\subsubsection{Comparison of MAP@K and NDCG}
+MAP@K and NDCG serve a similar purpose, as defined in \autoref{sec:evaluationmetrics}.
+For the predicted top-$N$ list we are not just interested in hitting relevant items, but also frontloading the list with these relevant items.
+These metrics serve to measure how well a method accomplishes this.
+As is evident from \autoref{graph:MAP10}, the MAP@K metric is very punishing, culminating in results of very small values approaching 0.
+NDCG on the other hand, \autoref{graph:NDCG10} tends to report more easily relatable results between 0 and 1 as an indicator.
+The reason for this discrepancy is in the way the calculation takes place.
+NDCG looks at whether or not items are frontloaded in a top-$N$ list whenever the top-$N$ list contains relevant items, unlike MAP@K, for which any list of top-$N$ contributes to the final results, rather than just those containing relevant items.
```

Comment:
```suggestion
NDCG looks at whether or not items are frontloaded in a top-$N$ list whenever the top-$N$ list contains relevant items, unlike MAP@N, for which any list of top-$N$ contributes to the final results, rather than just those containing relevant items.
```